fix: what I have observed, almost all typo

dev-itsheng opened the pull request:
# PR Details

Recently, I tried to understand front-end engineering, read the code of Modern.js, learned a lot, and fixed some problems I found.

<!--- Provide a general summary of your changes in the Title above -->

## Description

<!--- Describe your changes in detail -->

## Related Issue

<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

## Motivation and Context

<!--- Why is this change required? What problem does it solve? -->

## How Has This Been Tested

<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] My code follows the code style of this project.
- [ ] I have updated changeset
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

dev-itsheng commented:
> Thank you for your pull request and welcome to our community. We require contributors to sign our Contributor License Agreement, and we don't seem to have you on file. In order for us to review and merge your code, please follow the instructions at https://bytedance.feishu.cn/docs/doccnDUWkuuoOzFGtzoeMuWpzKc

I don't have permission to this linked document (probably because I'm not a ByteDance employee yet), please tell me how to sign the CLA, thanks.

chenjiahan commented:
Thanks for your PR, we are looking at the CLA issue, please wait for our feedback.

Also, the `Lint` and `Unit Test` CI are failed, please fix the CI.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/modern-js-dev/modern.js?pullRequest=983) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/modern-js-dev/modern.js?pullRequest=983) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: chenjiahan<br/>:x: dev-itsheng<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/modern-js-dev/modern.js?pullRequest=983) it.</sub>

chenjiahan commented:
@dev-itsheng We have updated the CLA bot, please sign online at [this link](https://cla-assistant.io/modern-js-dev/modern.js?pullRequest=983).

dev-itsheng commented:
> Thanks for your PR, we are looking at the CLA issue, please wait for our feedback.
> 
> Also, the `Lint` and `Unit Test` CI are failed, please fix the CI.

Due to time reasons, I only unit-tested the places I modified, not all of the unit tests in the project, so I didn't notice that there was a test that didn't belong to my changes that failed.

It goes like this:

<img width="1196" alt="image" src="https://user-images.githubusercontent.com/10683193/167064424-e53cddce-65f6-4d15-affb-36fc2029d34a.png">

source code:

```typescript
describe('plugin-nocode', () => {
  it('expect add style-loader to webpack config in nocode plugin', () => {
    try {
      userConfig._raw = userConfig;
      userConfig.source.include = ['query-string'];
      const config = new BaseWebpackConfig(
        appContext as IAppContext,
        userConfig,
      ).config();

      const hasStyleLoader = config?.module?.rules?.[1]?.oneOf.findIndex(
        (rule: RuleSetRule) => {
          return (
            Array.isArray(rule.use) &&
            typeof rule.use[0] === 'object' &&
            rule?.use?.[0]?.loader === 'style-loader'
          );
        },
      );
      expect(hasStyleLoader).toBeLessThan(0);
      handleWebpackConfig(config, {
        isDev: true,
        appDirectory: appContext.appDirectory,
        umdEntryFile: './',
      });

      const hasStyleLoaderAfterHandled =
        config?.module?.rules?.[1]?.oneOf.findIndex((rule: RuleSetRule) => {
          return (
            Array.isArray(rule.use) &&
            typeof rule.use[0] === 'object' &&
            rule.use[0].loader === 'style-loader'
          );
        });

      // ---------------------------------------------
      // this line, `hasStyleLoaderAfterHandled` is -1
      // ---------------------------------------------
      expect(hasStyleLoaderAfterHandled).toBeGreaterThanOrEqual(0);
    } catch (e) {
      console.error(e);
    }
  });
});
```

I looked at the changes related to this file and found that there was indeed one critical change that affected this test, see https://github.com/modern-js-dev/modern.js/pull/969, in `packages/cli/plugin-nocode/src/compiler/index.ts`, he change `'style-loader'` to `require.resolve('style-loader')`, but not modify the unit test:

```diff
  (webpackConfig?.module?.rules as RuleSetRule[])?.[1]?.oneOf?.forEach(rule => {
    if (
      Array.isArray(rule.use) &&
      typeof rule.use[0] === 'object' &&
      (rule.use[0]?.loader || '').includes('mini-css-extract-plugin')
    ) {
-      rule.use[0].loader = 'style-loader';
+      rule.use[0].loader = require.resolve('style-loader');
    }
  });
```

I have modified the unit test and made it pass:

```diff
describe('plugin-nocode', () => {
  it('expect add style-loader to webpack config in nocode plugin', () => {
    try {
      userConfig._raw = userConfig;
      userConfig.source.include = ['query-string'];
      const config = new BaseWebpackConfig(
        appContext as IAppContext,
        userConfig,
      ).config();

      const hasStyleLoader = config?.module?.rules?.[1]?.oneOf.findIndex(
        (rule: RuleSetRule) => {
          return (
            Array.isArray(rule.use) &&
            typeof rule.use[0] === 'object' &&
-           rule?.use?.[0]?.loader === 'style-loader'
+           rule?.use?.[0]?.loader === require.resolve('style-loader')
          );
        },
      );
      expect(hasStyleLoader).toBeLessThan(0);
      handleWebpackConfig(config, {
        isDev: true,
        appDirectory: appContext.appDirectory,
        umdEntryFile: './',
      });

      const hasStyleLoaderAfterHandled =
        config?.module?.rules?.[1]?.oneOf.findIndex((rule: RuleSetRule) => {
          return (
            Array.isArray(rule.use) &&
            typeof rule.use[0] === 'object' &&
-           rule?.use?.[0]?.loader === 'style-loader'
+           rule?.use?.[0]?.loader === require.resolve('style-loader')
          );
        });
      expect(hasStyleLoaderAfterHandled).toBeGreaterThanOrEqual(0);
    } catch (e) {
      console.error(e);
    }
  });
});
```

chenjiahan commented:
Yeah, this case is failed due to #969, thanks for you fix~

dev-itsheng commented:
I looked at some Lint errors, tried to fix some, others unsure turned off with `eslint-disable`

chenjiahan commented on packages/review/testing/src/config/testConfigOperator.ts:
```diff
@@ -31,22 +31,19 @@ class TestConfigOperator {
 
-  public mergeJestConfig(commingConfig: JestConfig) {
-    this._jestConfig = merge({}, this._jestConfig, commingConfig);
+  public mergeJestConfig(commonConfig: JestConfig) {
```
I think `commingConfig` is not a typo, it means `newConfig`

dev-itsheng replied:
So it's actually `comingConfig`.

But I still find the word weird, not even `newConfig` is appropriate, maybe we should use `userConfig`?

dev-itsheng replied:
After thinking about it, if I were to design a `merge`, I would call the incoming new parameter `source` and the original parameter `target`.

chenjiahan replied:
`source` and `target` is fine

dev-itsheng replied:
OK, I change `commonConfig` to `sourceConfig`.

chenjiahan commented on website/i18n/zh-cn/docusaurus-theme-classic/footer.json:
```diff
@@ -25,7 +25,7 @@
   "link.item.label.专题详解": {
     "message": "专题详解",
-    "description": "The label of footer link with label=专题详解 linking to /docs/guides/features/runtime/code-spliting"
+    "description": "The label of footer link with label=专题详解 linking to /docs/guides/features/runtime/code-splitting"
```
The `code-spliting.md` should be renamed too.

dev-itsheng replied:
Yes, I renamed it.

chenjiahan commented on packages/cli/core/compiled/ajv-keywords/ajv/dist/types/jtd-schema.d.ts:
```diff
@@ -34,7 +34,7 @@ export declare type SomeJTDSchemaType = (// ref
     [K in keyof T]-?: undefined extends T[K] ? never : K;
 }[keyof T];
-/** optional or undifined-able keys of an object */
+/** optional or undefined-able keys of an object */
```
`compiled` directory contains the code of pre-bundled dependencies, we should not manually modify the code in this directory.

dev-itsheng replied:
Yes, I reversed the changes to the three files in the `compiled` directory.

chenjiahan commented:
There are some ESLint issues in old code that cause CI to fail, this Pull Request will be merged first and the ESLint issues will be fixed later.
